### ISSUE TITLE
Inconsistency between LightGBM and XGBoost handling of column names containing <, >, and other unaccepted symbols.

### ISSUE THREAD

**Issue by angela97lin** [alteryx/evalml#1421]
LightGBM does:
```
    def _encode_categories(self, X, fit=False):
        X2 = pd.DataFrame(copy.copy(X))
        # encode each categorical feature as an integer
        X2.columns = np.arange(len(X2.columns))
        # necessary to wipe out column names in case any names contain symbols ([, ], <) which LightGBM cannot properly handle
        cat_cols = X2.select_dtypes(categorical_dtypes).columns
```

XGBoost does:
```
        # rename column names to column number if input is a pd.DataFrame in case it has column names that contain symbols ([, ], <) that XGBoost cannot properly handle
        if isinstance(X, pd.DataFrame):
            X = _rename_column_names_to_numeric(X)
```

For consistency, we should stick to one and use the implementation for both estimators.

**Comment by angela97lin** [comment]
Will be addressed by #1423

**Comment by angela97lin** [comment]
Closed by #1423
